lib/chef/resource/windows_ad_join.rb

```diff
@@ -52,6 +52,9 @@ class WindowsAdJoin < Chef::Resource
                description: "Controls the system reboot behavior post domain joining. Reboot immediately, after the Chef run completes, or never. Note that a reboot is necessary for changes to take effect.",
                default: :immediate
 
+      property :newname, String,
```

Comment:
Please can we follow the Ruby convention here, i.e. rename to `:new_name`

Reply:
right we discussed this, sorry windows habits